text/rule.go

```diff
@@ -15,28 +15,221 @@
 package text
 
 import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
 	"regexp"
 
 	engine "github.com/ZupIT/horusec-engine"
 )
 
+// MatchType represents the possibles match types of the engine
 type MatchType int
 
+// NotMatch will report any file that don't match the regex expressions
+// OrMatch for each regex that match will report a vulnerability
+// AndMatch need that all regex expressions match to report the vulnerability, it will get the first regex expression
+// the use as base to the reported vulnerability
+// Regular do the exact same thing as OrMatch, will be depreciated in the future to simplify use
```

Comment:
I think that would be better put the documentation above each enum
```go
const (
	// OrMatch for each regex that match will report a vulnerability
	OrMatch MatchType = iota

	// Regular do the exact same thing as OrMatch, will be depreciated in the future to simplify the use
	Regular

	// NotMatch will report any file that don't match the regex expressions
	NotMatch

	// AndMatch need that all regex expressions match to report the vulnerability, it will get the
	// first regex expression the use as base to the reported vulnerability
	AndMatch
)
```

Reply:
Done!